arp-speaker/main.go

```diff
@@ -62,32 +58,32 @@ func (c *controller) SetBalancer(name string, svc *v1.Service, eps *v1.Endpoints
 
 	if len(svc.Status.LoadBalancer.Ingress) != 1 {
 		glog.Infof("%s: no IP allocated by controller", name)
-		return c.deleteBalancer(name, "no IP allocated by controller")
+		return c.DeleteBalancer(name)
```

Comment:
I'm not super happy about losing the reason field here, it was pretty vital when debugging why services are not working correctly - otherwise all you know is that your service got deleted, but you don't know which of the 6-7 codepaths decided to delete it.

I think we should try to keep the reason around. And based on that... I'm wondering if this change actually makes sense, because all we're doing is making a method public and removing one "if svc == nil { ... }"... WDYT?

Reply:
I'm fine with leaving things as-is.